tests/languages.rs

```diff
@@ -0,0 +1,440 @@
+mod shared;
```

Comment:
I split the tests into multiple files since the one file was getting pretty long

From what I could tell in order to accomplish that you have to make them new separate "top-level" integration tests (under `tests/`)? Which seems fine

And per some guides extracted the shared stuff into a shared module that also sits under `tests/`